apps/papra-server/src/modules/documents/storage/drivers/az-blob/az-blob.storage-driver.ts

```diff
@@ -0,0 +1,37 @@
+import { BlobServiceClient, StorageSharedKeyCredential } from "@azure/storage-blob";
+import { Readable } from 'node:stream';
+
+import { defineStorageDriver } from '../drivers.models';
+
+export const AZ_BLOB_STORAGE_DRIVER_NAME = 'azure-blob' as const;
+
+export const azBlobStorageDriverFactory = defineStorageDriver(async ({ config }) => {
+  const { accountName, accountKey, containerName } = config.documentsStorage.drivers.azureBlob;
+
+  const blobServiceClient = new BlobServiceClient(`https://${accountName}.blob.core.windows.net`, new StorageSharedKeyCredential(accountName, accountKey));
+
+  return {
+    name: AZ_BLOB_STORAGE_DRIVER_NAME,
+    saveFile: async ({ file, storageKey }) => {
+      const containerClient = blobServiceClient.getContainerClient(containerName);
+      await containerClient.uploadBlockBlob(storageKey, Readable.fromWeb(file.stream()), file.size);
+      return { storageKey };
+    },
+    getFileStream: async ({ storageKey }) => {
+      const containerClient = blobServiceClient.getContainerClient(containerName);
+      const blockBlobClient = containerClient.getBlockBlobClient(storageKey);
+      const { readableStreamBody } = await blockBlobClient.download();
+
+      if (!readableStreamBody) {
+        throw new Error('File not found or has no content');
```

Comment:
I've added `createFileNotFoundError` in #264 (you need to rebase) to ensure same error across drivers

```suggestion
        throw createFileNotFoundError();
```